gramjs/client/TelegramClient.js

```diff
@@ -341,13 +354,13 @@ class TelegramClient {
             return this
         }
         if (args.code == null && !args.botToken) {
-            throw new Error('Please pass a promise to the code arg')
+            throw new Error('Please pass a string or a promise to the code arg')
```

Comment:
How would passing a string even work ?

Reply:
[`interactiveLogin.js`](https://github.com/gram-js/gramjs/pull/25/files#diff-8c9a90a6c2e45d471e7bc7a4c46b8d61) actually shows this. Granted this could also be done by going the promise route and just providing the `questionAsync` promise as the code argument.

Reply:
start already calls sendCodeRequest tho so you would end up with 2 codes ?

Reply:
Well this code is actually outdated, tbh idk how it's in the PR since `TelegramClient.js` doesn't exist anymore. If you check [`BaseClient.js`](https://github.com/gram-js/gramjs/pull/25/files#diff-06725ba56255eaa84e8fb65bc892f261) line 347 (since it won't let me link to a specific line for some reason) I'm doing a check to see if code is a function, if it is we so the `sendCodeRequest` and then await the code promise, otherwise we skip over `sendCodeRequest` and just pass the code value directly into the `signUp`/`signIn` call. I'm sure I could improve the code somewhat, but it is working.

Reply:
oh that's cool.  I didn't notice that.